File: src/turtle_control/scripts/move_turtle_in_circle.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String
from geometry_msgs.msg import Twist
import math

if __name__ == '__main__':
    rospy.init_node('move_turtle_in_circle')

    pub = rospy.Publisher("/turtle1/cmd_vel", Twist, queue_size=10)

    rate = rospy.Rate(2)

    vals = range(1,6)
    counter = 0

    while not rospy.is_shutdown():
        msg = Twist()
        msg.linear.x = 5

        rospy.loginfo(vals[counter])

        msg.angular.z = math.pi *vals[counter]
        pub.publish(msg)
        rate.sleep()
        counter += 1

        if counter == 5:
            counter = 0

    rospy.loginfo("Node was stopped!")

    

File: src/turtle_control/src/move_turtle_in_circle.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <geometry_msgs/Twist.h>
#include <math.h>

int main(int argc, char **argv){
    
    ros::init(argc, argv, "move_turtle_in_circle");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 10);

    ros::Rate rate(2);

    while (ros::ok()){
        geometry_msgs::Twist msg;
        msg.linear.x = 5;
        msg.angular.z = M_PI * 2;
        pub.publish(msg);
        rate.sleep();
    }
    
}